src/components/network-map-tab.js

```diff
@@ -209,43 +209,159 @@ export const NetworkMapTab = ({
         []
     );
 
+    const getEquipmentsNotFoundIds = useCallback(
+        (foundEquipmentPositions, allEquipments) => {
+            const foundEquipmentsIds = Array.from(
+                foundEquipmentPositions.keys()
+            );
+
+            const notFoundEquipmentsIds = allEquipments
+                .filter((s) => !foundEquipmentsIds.includes(s.id))
+                .map((s) => s.id);
+
+            return notFoundEquipmentsIds;
+        },
+        []
+    );
+
+    const getMissingEquipmentsPositions = useCallback(
+        (notFoundEquipmentsIds, fetchEquipmentCB) => {
+            if (notFoundEquipmentsIds.length === 0) {
+                return Promise.resolve([]);
+            }
+
+            return fetchEquipmentCB(
+                studyUuid,
+                currentNode?.id,
+                notFoundEquipmentsIds
+            );
+        },
+        [studyUuid, currentNode?.id]
+    );
+
     const loadMapGeoData = useCallback(() => {
```

Comment:
I found this method a bit intimidating at first, maybe break it up into two other methods called something like : loadAllGeoData and loadMissingGeoData ?

Reply:
Yes it's too big and the logic is not obvious as it's too long. I split it